cmd/agent/main.go

```diff
@@ -485,3 +510,75 @@ func reconcileKubelet(ctx context.Context) (bool, error) {
 	}
 	return false, nil
 }
+
+// getIPAddress gets the machines primary outbound IP address by making a UDP
+// request to a random (non-existent) host and taking the local address
+// used to make the request.
+func getIPAddress() string {
+	conn, err := net.Dial("udp", "1.3.3.7:1")
+	if err != nil {
+		return ""
+	}
+	defer conn.Close()
+	return conn.LocalAddr().(*net.UDPAddr).IP.String()
+}
+
+func KubeletNeedsNewCertificate(headers map[string][]string) error {
```

Comment:
Nit: Might be beyond the scope of this PR, but maybe `http.Header` could be used here. I think `headers.Set(RegenerateKubeletCertificate, "true")` would be cleaner than `		headers[RegenerateKubeletCertificate][0] = "true"`

Reply:
Good point!